### PR TITLE
Spruce up README and add CONTRIBUTING

### PR DESCRIPTION
**Pull request by JonDum**
All righty y'all https://github.com/webpack/webpack/issues/908

See how the readme looks on my fork: https://github.com/JonDum/webpack

I had the idea of adding some of the loaders to the readme because it gives a good idea about the types of things webpack can do, but I think there might be too many now haha.

There's still also the issue of organizing and tagging Issues as they come in.

### CODE REVIEW

**Comment by andreypopp** [comment]
nitpick: I'd move jsx-loader out of templating.... into dialects maybe?

File: README.md [hunk 1]
```diff
@@ -30,33 +31,112 @@ global:
...
+Webpack can do many optimizations to **reduce the output size of your
+javascript** by deduplicating frequently used modules, minifying, and giving
+you full control of what is loaded initially and what is loaded at runtime
+through code splitting. It can also can make your code chunks **cache
+friendly** by using hashes. 
 
+[Optimization documentation](http://webpack.github.io/docs/optimization.html)
 
 
 # A small example of what's possible
```
Comment (andreypopp) [inline README.md]: I think this small example is rather overwhelming... Readers already can appreciate webpack flexibility from the plugins/loaders listings above.

File: README.md [hunk 2]
```diff
@@ -30,33 +31,112 @@ global:
...
+* [`eslint`](https://github.com/MoOx/eslint-loader): PreLoader for linting code using ESLint.
+* [`jshint`](https://github.com/webpack/jshint-loader): PreLoader for linting code.
+* [`jscs`](https://github.com/unindented/jscs-loader): PreLoader for style checking.
+* [`injectable`](https://github.com/jauco/webpack-injectable): Allow to inject dependencies into modules
+* [`transform`](https://github.com/webpack/transform-loader): Use browserify transforms as loader.
+* [`image-size`](https://github.com/patcoll/image-size-loader): Loads an image and returns its dimensions and type
+
+For the full list of loaders, please see [list of loaders](http://webpack.github.io/docs/list-of-loaders.html)
 
 ## Support
```
Comment (andreypopp) [inline README.md]: There's subsection *support* in the paragraph above and it is kinda confusing... I think it's better to call the section *Various module formats* or something like that.

File: README.md [hunk 3]
```diff
@@ -30,33 +31,112 @@ global:
...
+* [`jade`](https://github.com/webpack/jade-loader): Loads jade template and returns a function
+* [`handlebars`](https://github.com/altano/handlebars-loader): Loads handlebars template and returns a function
+* [`ractive`](https://github.com/rstacruz/ractive-loader): Pre-compiles Ractive templates for interactive DOM manipulation
+* [`jsx`](https://github.com/petehunt/jsx-loader): Transform jsx code for [React](http://facebook.github.io/react/) to js code.
+* [`em`](https://github.com/yoshdog/emblem-loader): Compiles [Emblem](http://emblemjs.com/) to Handlebars.js
+* [`ejs`](https://github.com/okonet/ejs-loader): Loads EJS ([underscore](http://underscorejs.org/#template)( templating engine) template and returns a pre-compiled function
+* [`markdown`](https://github.com/peerigon/markdown-loader): Compiles Markdown to HTML
+* [`ng-cache`](https://github.com/teux/ng-cache-loader): Puts HTML partials in the Angular's $templateCache
+
+**styling**
```
Comment (andreypopp) [inline README.md]: I think to highlight that webpack can produce `css` assets we can add [extract-text-webpack-plugin](https://github.com/webpack/extract-text-webpack-plugin) here.

File: README.md [hunk 4]
```diff
@@ -30,33 +31,112 @@ global:
...
+within webpack itself use this plugin interface. This makes webpack very
+**flexible**.
 
 ## Performance
 
-webpack uses async I/O and has multiple caching levels. This makes webpack fast and incredibly **fast** on incremental compilations.
+Webpack uses async I/O and has multiple caching levels. This makes webpack fast
+and incredibly **fast** on incremental compilations.
 
 ## Loaders
```
Comment (andreypopp) [inline README.md]: Maybe we can renamed section to *Plugins and loaders* or even *Modular design* and mention few more useful webpack goodies like [webpack-dev-server](https://github.com/webpack/webpack-dev-server) and [webpack-dev-middleware](https://github.com/webpack/webpack-dev-middleware) here.

**Comment by andreypopp** [comment]
I think README could also mention Hot Module Reloading as a feature.

File: README.md (hunk 4 above)
Comment (andreypopp) [inline README.md]: I think we can also provide a link to ["webpack" keyword on npm](https://www.npmjs.com/browse/keyword/webpack) for further results besides highlighted plugins/loaders.

File: README.md [hunk 5]
```diff
@@ -30,33 +31,112 @@ global:
...
+
+**packaging**
+* [`file`](https://github.com/webpack/file-loader): Emits the file into the output folder and returns the (relative) url.
+* [`url`](https://github.com/webpack/url-loader): The url loader works like the file loader, but can return a Data Url if the file is smaller than a limit.
+* [`image`](https://github.com/tcoopman/image-webpack-loader): Compresses your images. Ideal to use together with `file` or `url`.
+* [`svgo-loader`](https://github.com/pozadi/svgo-loader): Compresses SVG images using [svgo](https://github.com/svg/svgo) library
+* [`baggage`](https://github.com/deepsweet/baggage-loader): Automatically require any resources related to the required one
+* [`polymer-loader`](https://github.com/JonDum/polymer-loader): Process HTML & CSS with preprocessor of choice and `require()` Web Components like first-class modules.
+
+**dialects**
```
Comment (hkal) [inline README.md]: Call this transpilers instead?

File: README.md [hunk 6]
```diff
@@ -30,33 +31,112 @@ global:
...
+* [`po`](https://github.com/dschissler/po-loader): Loads a PO gettext file and returns JSON
+
+**support**
+* [`mocha`](https://github.com/webpack/mocha-loader): do tests with mocha in browser or node.js
+* [`eslint`](https://github.com/MoOx/eslint-loader): PreLoader for linting code using ESLint.
+* [`jshint`](https://github.com/webpack/jshint-loader): PreLoader for linting code.
+* [`jscs`](https://github.com/unindented/jscs-loader): PreLoader for style checking.
+* [`injectable`](https://github.com/jauco/webpack-injectable): Allow to inject dependencies into modules
+* [`transform`](https://github.com/webpack/transform-loader): Use browserify transforms as loader.
+* [`image-size`](https://github.com/patcoll/image-size-loader): Loads an image and returns its dimensions and type
```
Comment (hkal) [inline README.md]: I think this section should be called `code quality`. Also, I am not clear on the purpose of the `image-size` loader.

File: README.md (hunk 1 above)
Comment (JonDum) [inline README.md]: Yea I'm gonna take another run through and remove a lot more of them.

Comment (JonDum) [inline README.md]: Quite possible. Let's get some more opinions on it.

**Comment by JonDum** [comment]
@andreypopp Could you take a stab at writing a section for HMR for me? I honestly haven't used that myself yet.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+
+## Submitting Changes
+
+* Push to your fork and submit a pull request.
+* Wait for feedback. We may suggest some changes or improvements or alternatives.
+
+Some things that will increase the chance that your pull request is accepted:
+
+* Write tests.
+* Follow our coding style.
```
Comment (hkal) [inline CONTRIBUTING.md]: There does not appear to be a rigorous coding guideline in place. However, there is an `.editorconfig` file that attempts to ensure whitespace formatting. Until there is something more defined around code style, telling people to use [editorconfig](http://editorconfig.org/) is the best bet, imo.

**Comment by JonDum** [comment]
Added a revision. Loving the feedback. I'm going to wait for some others to chime in before doing more now.

**Comment by sokra** [comment]
@JonDum Still reading... Could you replace `Webpack` with `webpack`? Lowercase is the correct spelling.

File: README.md (hunk 1 above)
Comment (jhnns) [inline README.md]: Same here. I never found that example convincing because it demonstrates already too much.

File: CONTRIBUTING.md (hunk 7 above)
Comment (jhnns) [inline CONTRIBUTING.md]: I've always included in our open source projects these lines:

> From opening a bug report to creating a pull request: every contribution is appreciated and welcome. If you're planing to implement a new feature or change the api please create an issue first. This way we can ensure that your precious work is not in vain.

I found it to be more satisfying for both sides if breaking or bigger changes are discussed in the first place.

Comment (JonDum) [inline CONTRIBUTING.md]: @jhnns +1 

That's a very nice way of saying it. I'll add it in.

**Comment by sokra** [comment]
Thanks...

**Pull request merged**